guidelines/application_requirements.md

```diff
@@ -0,0 +1,84 @@
+# Guidelines for Application Requirements
+
+Authors: chasm@google.com, jlewi@google.com, kam.d.kasravi@intel.com
+
+## Objective
+
+Define requirements and recommendations aimed at graduating applications (e.g. controller and web apps) in Kubeflow to 1.0.
+
+## Scope
+
+This document is intended to cover applications that are deployed in the Kubernetes cluster and in particular:
+
+1. Custom resources and controllers
+1. Web applications
+
+This document is not intended to cover SDKs (e.g. fairing) or client side tools (like kfctl).
+
+## Goals
+
+The purpose of this doc is to:
+
+1. Establish criterion for graduating a Kubeflow application to a supported Kubeflow application; i.e. 1.0
```

Comment:
From discussions earlier today: When we talk about moving a Kubeflow application to v1.0, there are actually two things involved:
* Moving the application itself to v1.0 of the application.
* Declaring the application to be a supported part of Kubeflow v1.x (or 2.x, etc).

I think we should make this clear in this doc, and also clarify whether these are the criteria for the app itself to move to v1.0 or for the app to be considered "Kubeflow 1.0+ compliant".